### ISSUE TITLE
Mustache loop with waitforimages?

### ISSUE THREAD

**Issue by m1ga** [alexanderdickson/waitForImages#65]
I have to following mustache loop:

~~~
<div class="grid">
    <div class="row">

        <script id="template" type="text/html">
            {{#items}}
            <div class="grid__item col-xs-4 col-md-3" data-type="{{type}}" data-module="{{ module }}" data-content="{{ content }}">
                <div class="grid__bl"></div>
                <div class="grid__br"></div>
                <div class="grid__tl"></div>
                <div class="grid__tr"></div>
                <div class="grid__image" style="background-image: url({{ image }});"></div>
                <div class="grid__header">
                    <h2 class="grid__title">{{ title }}</h2>
                </div>
            </div>
            {{/items}}
        </script>
    </div>
</div>
~~~

and grid__image will contain the images I want to load.
In my js file i have:

~~~
$(function() {
        $.waitForImages.hasImgProperties = ['backgroundImage'];

        var template = $('#template').last().html();
        Mustache.parse(template); 
        var rendered = Mustache.render(template, {
            items: data
        });
        $('.row').last().html(rendered);

        $('.grid').waitForImages(function() {
            overlayLoaded();
        }, function(loaded, total) {
            console.log(parseInt(loaded / total * 100) + '% loaded.');
        });
});
~~~

but i'm not getting any console log output and the overlayLoaded() function is called before all images are loaded (div is outside and should move in once all images are loaded. But it will move in before)

**Comment by alexanderdickson** [comment]
The most obvious I could see is you set `hasImgProperties` to only check background images. But you don't set `waitForAll` to look at backgorund images too. Provide a third argument to `waitForImages()`, which is `true`.

If that doesn't work, are you able to demonstrate the issue on jsfiddle or similar?

Things I would check

- is `.grid` found?
- at the time `waitForImages()` is called, are there any images found by the jQuery `find()` code in the plugin?
- have you stepped through the code using the browser debugger to see where it took a wrong turn?

Also, `loaded` and `total` are already Numbers, if you want an integer, a better function would be to use `Math.floor()`. `parseInt()` is designed for something else (parsing numbers from strings).

**Comment by m1ga** [comment]
Thanks for your help! The third parameter was missing and after adding it worked like a charm.

**Comment by alexanderdickson** [comment]
Cool, is there any advice for making the documentation clearer?

**Comment by m1ga** [comment]
Perhaps an example for background-images would help since its a common use case i think.

I got confused by 
~~~
(by default, it looks at the background-image, list-style-image, border-image, border-corner-image and cursor properties)
~~~
I thought that even if I don't set the third parameter it would be default. But I think it means that if I set it to true it will be a default and I don't to change the hasImgProperties, right?
